src/Bellatrix.Playwright/components/core/Component.DefaultActions.cs

```diff
@@ -0,0 +1,235 @@
+﻿// <copyright file="Component.DefaultActions.cs" company="Automate The Planet Ltd.">
+// Copyright 2024 Automate The Planet Ltd.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+// <author>Miriam Kyoseva</author>
+// <site>https://bellatrix.solutions/</site>
+
+using System.Diagnostics;
+using System.Drawing;
+using Bellatrix.Playwright.Events;
+using Microsoft.VisualStudio.Services.WebApi;
+
+namespace Bellatrix.Playwright;
+
+public partial class Component
+{
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+    public Point Location => new((int)WrappedElement.BoundingBox().X, (int)WrappedElement.BoundingBox().Y);
+
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+    public Size Size => new((int)WrappedElement.BoundingBox().Width, (int)WrappedElement.BoundingBox().Height);
+
+    // TODO: Will GetCssValue() work?
+    public string GetCssValue(string propertyName) => WrappedElement.Evaluate($"element => window.getComputedStyle(element).getPropertyValue('{propertyName}')").GetValueOrDefault().ToString();
+
+    protected virtual void DefaultSetAttribute(string attributeName, string attributeValue)
+    {
+        SettingAttribute?.Invoke(this, new ComponentActionEventArgs(this));
+
+        WrappedElement.Evaluate($"el => el.{attributeName} = '{attributeValue}'");
+
+        AttributeSet?.Invoke(this, new ComponentActionEventArgs(this));
+    }
+
+    protected virtual string GetClassAttribute()
+    {
+        return string.IsNullOrEmpty(GetAttribute("class")) ? null : GetAttribute("class");
+    }
+
+    internal void DefaultClick(EventHandler<ComponentActionEventArgs> clicking, EventHandler<ComponentActionEventArgs> clicked)
+    {
+        clicking?.Invoke(this, new ComponentActionEventArgs(this));
+
+        // Should we first perform a normal click? or at least provide this option
```

Comment:
the "normal" click might work fine in playwright, but in Selenium it is performing some additional checks before clicking that are giving false-positive results in some cases. Such as Element not clickable at point x,y other element will recieve the click. I would encourage you to try with the native implementation and move to JSClick just in case it has some issues.